### PR TITLE
refactored DatetimeInfoExtractor to improve readability/condense

### PR DESCRIPTION
**Pull request by limlam96**
Refactored DatetimeInfoExtractor for readability reasons

### CODE REVIEW

File: tubular/dates.py [hunk 1]
```diff
@@ -1033,37 +1023,17 @@ def _map_values(self, value: float, interval: str) -> str:
...
+            if value.is_integer():
+                value = int(value)
 
-        if np.isnan(value):
-            return np.nan
+        if isinstance(value, int) and value in self.RANGE_TO_MAP[include_option]:
+            return self.inverted_datetime_mappings[include_option][value]
 
-        return mappings[interval][value]
+        msg = f"{self.classname()}: value for {include_option} mapping in self._map_values should be an integer value in {min(self.RANGE_TO_MAP[include_option])}-{max(self.RANGE_TO_MAP[include_option])}"
```
Comment (davidhopkinson26) [inline tubular/dates.py:1035]: S/N - feel like this might read better as an `else` even if not required for the logic to work.

**Review by davidhopkinson26 — APPROVED** [review]
Great improvement thanks!  Reads more clearly, as advertised.  One tiny suggestion but happy for you to ignore and merge as is if you prefer.

File: tubular/dates.py (hunk 1 above)
Comment (limlam96) [reply tubular/dates.py:1035]: personally I don't disagree, but this has been beaten into me by ruff rules that we've added to HomeX haha https://docs.astral.sh/ruff/rules/superfluous-else-raise/

If we think we might turn on more of these in future maybe would leave?

**Comment by davidhopkinson26** [comment]
> Spotted I hadn't updated changelog so pushed again :)

thanks

File: tubular/dates.py (hunk 1 above)
Comment (davidhopkinson26) [reply tubular/dates.py:1035]: I bow to the authority of ruff in that case.  Prefer to leave stylistic choices to a linter where possible.

**Review by davidhopkinson26 — APPROVED** [review]

**Pull request merged**